provisioning/transport/amqp/src/AmqpAuthStrategyX509.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Transport
+{
+    internal class AmqpAuthStrategyX509 : AmqpAuthStrategy
+    {
+        SecurityClientHsmX509 _security;
+
+        public AmqpAuthStrategyX509(SecurityClientHsmX509 security)
+        {
+            _security = security;
+        }
+
+        public override AmqpSettings CreateAmqpSettings(string linkendpoint)
+        {
+            return new AmqpSettings();
+        }
+
+        public override Task OpenConnectionAsync(AmqpClientConnection connection, TimeSpan timeout, bool useWebSocket)
+        {
+            // TODO: install certificate chain.
```

Comment:
>// TODO: install certificate chain. [](start = 12, length = 35)

What's this for?

Reply:
It's actually done by the `ProvisioningDeviceClient.Create` factory at the moment but the code will be moved into the SecurityClientX509.